app/gui2/src/components/ComponentBrowser/input.ts

```diff
@@ -392,6 +419,13 @@ export function useComponentBrowserInput(
 
   function codeToBeInserted(entry: SuggestionEntry): string {
     const ctx = context.value
+    if (ctx.type === 'insert' && ctx.position === 0 && sourceNode.value) return entry.name
+    if (
+      ctx.type === 'changeIdentifier' &&
+      ctx.identifier.inner.whitespaceStartInCodeParsed === 0 &&
+      sourceNode.value
+    )
+      return entry.name
```

Comment:
I think we could have fewer exceptions of this kind, if we use a `fullAst` computed value, i.e. parsed `${sourceNode}.${text}`. Then in places where we're returning new input, we could just skip `${sourceNode}.` prefix. As a bonus, the `fullAst` could be published and used to create a new node.

But I do not treat it as a blocker, as it might be harder than I think. Or not actually cleaner.